client/src/components/QueryComponent/QueryComponent.test.js

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import { render, unmountComponentAtNode } from "react-dom";
+import { act, Simulate } from "react-dom/test-utils";
+import { findByText, findByLabelText, waitFor, findByTestId, fireEvent, screen } from "@testing-library/react";
+import user from '@testing-library/user-event';
+import PubSub from "pubsub-js";
+import QueryComponent from "./QueryComponent";
+import { configService } from "../../services/ConfigService";
+import initIcons from "../../services/IconService/IconService";
+import { settingsService } from "../../services/SettingsService";
+
+initIcons();
+
+jest.mock("../../services/ConfigService");
+jest.mock("../../services/SettingsService");
+
+const mockQueriesResponse = [{ name: "FloorQuery", query: "SELECT * FROM digitaltwins T where T.$dtId in ['Floor01', 'Floor04']"}];
+
+let container = null;
+beforeEach(() => {
+  container = document.createElement("div");
+  document.body.appendChild(container);
+});
+
+afterEach(() => {
+  PubSub.clearAllSubscriptions();
+  unmountComponentAtNode(container);
+  container.remove();
+  container = null;
+});
+
+// console.log(screen.debug(null,20000));
+
+
+test("load saved query", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  settingsService.queries = mockQueriesResponse;
+  act(() => {render(<QueryComponent/>, container);});
+  
+  const savedDropdown = await screen.getByTestId("savedQueryDropdown");
+
+  act(() => {
+    savedDropdown.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+  });
+
+  const dropdownOption = await screen.findByTitle("FloorQuery");
+
+  act(() => {
+    dropdownOption.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+  });  
+  
+  const queryInput = await screen.getByTestId("queryInput");
+
+  expect(queryInput.value).toBe("SELECT * FROM digitaltwins T where T.$dtId in ['Floor01', 'Floor04']");
+
+})
+
+test("save query", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  settingsService.queries = mockQueriesResponse;
+  act(() => {render(<QueryComponent/>, container);});
+
+  const savedDropdown = await screen.getByTestId("savedQueryDropdown");
+
+  act(() => {
+    savedDropdown.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+  });
+  
+  let dropdownOption = await screen.queryAllByTitle("TestQuery");
+
+  expect(dropdownOption.length).toBe(0);
```

Comment:
I'm a bit confused here. If settingsService.queries is set to mockQueriesResponse, shouldn't it start with 1 saved query?

Reply:
Actually you are Right the Dropdown have 2 Options but the query screen.queryAllByTitle("TestQuery"); is bringing just the ones matching with the TestQuery Title so that is why we have 0, the Title for the Mocked Query is  FloorQuery